master/buildbot/status/worker.py

```diff
@@ -41,6 +42,30 @@ def __init__(self, name):
         self.graceful_callbacks = []
         self.pause_callbacks = []
         self.connect_times = []
+        self.info = Properties()
+
+    def setInfo(self, info):
+        self.admin = bytes2unicode(info.get("admin"))
+        self.host = bytes2unicode(info.get("host"))
+        self.access_uri = info.get("access_uri", None)
+        if self.access_uri:
+            self.access_uri = bytes2unicode(self.access_uri)
+        self.version = bytes2unicode(info.get("version", b"(unknown)"))
+
+        # store everything as Properties
+        for k, v in info.items():
+            # be careful about few items.
+            if k == 'numcpus':
+                self.info.setProperty(bytes2unicode(k), v, "Worker")
+            elif k in ('environ', 'worker_commands'):
+                continue
+            elif v is None:
+                self.info.setProperty(bytes2unicode(k), None, "Worker")
```

Comment:
I think we need a test that covers this case.

Reply:
i'll add these